src/Rules/Validation/ScopeValidationMethods.php

```diff
@@ -0,0 +1,91 @@
+<?php declare(strict_types=1);
+
+namespace Hihaho\PhpstanRules\Rules\Validation;
+
+use Illuminate\Support\Collection;
+use Illuminate\Support\Stringable;
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Type\ObjectType;
+use ReflectionClass;
+use ReflectionException;
+use ReflectionIntersectionType;
+use ReflectionMethod;
+use ReflectionNamedType;
+use ReflectionParameter;
+use ReflectionUnionType;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Expr\MethodCall>
+ */
+abstract class ScopeValidationMethods implements Rule
+{
+    abstract public function getNodeType(): string;
+
+    abstract public function processNode(Node $node, Scope $scope): array;
+
+    /**
+     * @phpstan-return ReflectionMethod[]
+     * @throws ReflectionException
+     */
+    protected function getClassMethods(Scope $scope): array
```

Comment:
the `Scope` object contains the PHP class that is being inspected. This method gathers all the methods that are present in that class